Sources/NostrSDK/DirectMessageEncrypting.swift

```diff
@@ -0,0 +1,170 @@
+//
+//  DirectMessageEncrypting.swift
+//  
+//
+//  Created by Joel Klabo on 8/10/23.
+//
+
+import Foundation
+import secp256k1
+import CommonCrypto
+import CryptoKit
+
+public enum DirectMessageEncryptingError: Error {
+    case pubkeyParsing
+    case sharedSecretCreation
```

Comment:
Can this be stated as the error that happened? What about the shared secret creation happened?